### PR TITLE
GEOPY-1450: Geoimage in geoh5py import tiff return black image

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1450 - Geoimage in geoh5py import tiff return black image**
Change the conversion from geoimage to grid, not converting data type anymore.
If the input image is a tiff, the data is automaticly duplicated as a grid

### CODE REVIEW

File: geoh5py/objects/geo_image.py [hunk 1]
```diff
@@ -371,15 +372,16 @@ def image(self, image: str | np.ndarray | BytesIO | Image.Image):
                 raise ValueError(f"Input image file {image} does not exist.")
 
             image = Image.open(image)
-
             # if the image is a tiff save tag information
             if isinstance(image, TiffImageFile):
                 self.tag = image
+                convert_to_grid = True
```
Comment (domfournier) [inline geoh5py/objects/geo_image.py:378]: Combine duplicate logic. All of these could be just if statements that converts nd.array or str or bytes to an image. Then do the checks for is `Image` and is `TiffImageFile` once

File: tests/geo_image_test.py [hunk 2]
```diff
@@ -586,3 +586,23 @@ def test_copy_from_extent_geoimage(tmp_path):
...
+    with Workspace.create(workspace_path) as workspace:
+        # load image
+        geoimage = GeoImage.create(workspace, name="test_area", image=str(image_path))
+
+        grid = workspace.get_entity("test_area_grid2d")[0]
+
+        assert all(
+            np.array(geoimage.image)[::-1].flatten()
+            == grid.get_data("band[0]")[0].values
+        )
```
Comment (domfournier) [inline tests/geo_image_test.py:608]: Can we add a check for multi-bands too?

**Review by domfournier — COMMENTED** [review]
Simple, that's great. See comments

File: tests/geo_image_test.py (hunk 2 above)
Comment (MatthieuCMira) [reply tests/geo_image_test.py:608]: PIL images can't handle >4 bands. 
That's why people are using Gdal

Comment (MatthieuCMira) [reply tests/geo_image_test.py:608]: Actually, I can play with more than 4 bands, but that's another level of complexity. 
I created an issue to refactor geoimage for tiff.

Comment (MatthieuCMira) [reply tests/geo_image_test.py:608]: I'll test for 4 bands here, and we will implement n bands later

Comment (domfournier) [reply tests/geo_image_test.py:608]: Ok doke

**Review by andrewg-mira — APPROVED** [review]

**Pull request merged**